cors.go

```diff
@@ -1 +1,126 @@
+// Copyright 2021 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package cors is a middleware that generates CORS headers for Flamego.
 package cors
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/flamego/flamego"
+)
+
+const anyDomain = "!*"
+
+// Options contains options for the cors.CORS middleware.
+type Options struct {
+	// Scheme may be http or https as accepted schemes or the '*' wildcard to accept any scheme. (default: "http")
+	Scheme string
+	// AllowDomain is a comma separated list of domains that are allowed to run CORS requests
+	// Special values are the  a single '*' wildcard that will allow any domain to send requests without
+	// credentials and the special '!*' wildcard which will reply with requesting domain in the 'access-control-allow-origin'
+	// header and hence allow requests from any domain *with* credentials. (default '*')
+	AllowDomain []string
+	// AllowSubdomain allowed subdomains of domains to run CORS requests. (default false)
+	AllowSubdomain bool
+	// Methods may be a comma separated list of HTTP-methods to be accepted. (default GET, POST, OPTIONS)
+	Methods []string
+	// MaxAgeSeconds may be the duration in secs for which the response is cached. (default 600)
+	MaxAgeSeconds int
```

Comment:
```suggestion
	MaxAge time.Duration
```

I think this is much easier to use in Go, and when we use it, we just call `MaxAge.Seconds()`